File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var hash = {}; // number, index
    
    for (var i = 0; i < nums.length; i++) {
        var x = nums[i];
        
        if (target - x in hash) {
            return [hash[target-x], i];
        } else {
            hash[x] = i;
        }
    }
    
};

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {} # value -> index
        i = 0
        l = len(nums)
        
        while i < l:
            diff = target - nums[i]
            if diff in map:
                return [map[diff], i]
            map[nums[i]] = i
            i += 1